lib/kubernetes-deploy/kubernetes_resource/persistent_volume_claim.rb

```diff
@@ -3,16 +3,76 @@ module KubernetesDeploy
   class PersistentVolumeClaim < KubernetesResource
     TIMEOUT = 5.minutes
 
+    def sync(cache)
+      super
+      @storage_classes = cache.get_all("StorageClass").map { |sc| StorageClass.new(sc) }
+    end
+
     def status
       exists? ? @instance_data["status"]["phase"] : "Not Found"
     end
 
     def deploy_succeeded?
-      status == "Bound"
+      return false if deploy_failed?
+      return true if status == "Bound"
+
+      # if the StorageClass has volumeBindingMode: WaitForFirstConsumer,
+      # it won't bind until after a Pod mounts it. But it must be pre-deployed,
+      # as the Pod requires it. So 'Pending' must be treated as a 'Success' state
+      if storage_class&.volume_binding_mode == "WaitForFirstConsumer"
+        status == "Pending" || status == "Bound"
+      end
     end
 
     def deploy_failed?
-      status == "Lost"
+      status == "Lost" || failure_message.present?
+    end
+
+    def failure_message
+      if storage_class_name.nil? && @storage_classes.count(&:default?) > 1
+        "PVC has no StorageClass specified and there are multiple StorageClasses" \
+        "with annoated as default. This is an invalid cluster configuration."
```

Comment:
Don't need the word `with`
```suggestion
        "annoated as default. This is an invalid cluster configuration."
```

Reply:
nit: also `annoated` -> `annotated`